docs/src/main/tut/typeclasses/eq.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: docs
+title:  "Eq"
+section: "typeclasses"
+source: "kernel/src/main/scala/cats/kernel/Eq.scala"
+scaladoc: "#cats.kernel.Eq"
+---
+
+# Eq
+
+Eq is an alternative to the standard Java `equals` method.
+It is defined by the single method `eqv`:
+
+```scala
+def eqv(x: A, y: A): Boolean
+```
+
+In Scala it's possible to compare any two values using `==` (which desugars to Java `equals`).
+This is because `equals` type signature uses `Any` (Java's `Object`) to compare two values.
+This means that we can compare two completely unrelated types without getting a compiler error.
+The Scala compiler may warn us in some cases, but not all, which can lead to some weird bugs.
+For example this code will raise a warning at compile time:
+
+
+```tut:book:fail
+42 == "Hello"
+```
+
+While this code will compile without a hitch:
+
+```tut:book
+"Hello" == 42
+```
+
+Ideally, Scala shouldn't let us compare two types that can never be equal.
+
+As you can probably see in the type signature of `eqv`, it is impossible to compare two values of different types,
+eliminating these types of bugs altogether.
+
+The `Eq` syntax package also offers some handy symbolic operators:
+
+```tut:book
+import cats.implicits._
+
+1 === 1
+
+"Hello" =!= "World"
+```
+
+Implementing `Eq` instances yourself for every data type might seem like huge drawback compared to only slight gains of typesafety.
+Fortunately for us, we have two great options. One option is to use inbuilt helper functions.
+Another option is to use a small library called [kittens](https://github.com/milessabin/kittens), which can derive a lot of type class instances for our data types including `Eq`.
+
+The first option using `Eq.fromUniversalEquals` only defers to `==` and works like this:
+
+```tut:book
+import cats.kernel.Eq
+import cats.implicits._
+
+
+case class Foo(a: Int, b: String)
+
+
+implicit val eqFoo: Eq[Foo] = Eq.fromUniversalEquals
+
+
+Foo(10, "") === Foo(10, "")
+```
+
+You can even go one step further by making this implicitly available for all `scala.Product`s (which include case classes).
+```tut:book
+
+case class Bar(a: Double, b: Int)
+
+implicit def eqProduct[A <: Product]: Eq[A] = Eq.fromUniversalEquals
```

Comment:
Sorry to disagree with a couple of other commenters, but personally I wouldn't be too inclined to include this example in the docs. A couple of significant difference between this and kittens derivation is that kittens will only derive the `Eq` instance when all types within the case class have `Eq` instances, and it will be consistent with those individual `Eq` instances, while this relies on universal equality so it won't necessarily.

I like the `fromUniversalEquals` method and use it on occasion, but I think it's a big leap to add this instance into implicit scope for _all_ `Product` types. I don't want to be a stick in the mud though, so if others want to go forward with this in the docs, that's okay.

Reply:
that's a really good point. It didn't come to my mind that fromUniversalEquals wont recursively use `Eq` for fields. Agree we should remove this example usage.